doc/testing.md

```diff
@@ -0,0 +1,25 @@
+# Testing
+
+## Unit Tests
+
+Write tests in files whose name ends with `_test.py` which ensures that they
+are picked up by `py.test` command. These test files will contain functions
+with function name matching `testXxx` pattern. Put these test files in the
+same package as the one being tested. This ensures that unexported functions
+can also be unit tested.
```

Comment:
Should this paragraph mention that tests will go in the test directory?

Reply:
done.